prime-router/src/main/kotlin/Report.kt

```diff
@@ -337,13 +337,27 @@ class Report {
         }
     }
 
-    fun filter(filterFunctions: List<Pair<JurisdictionalFilter, List<String>>>, receiver: Receiver): Report {
+    fun filter(
+        filterFunctions: List<Pair<JurisdictionalFilter, List<String>>>,
+        receiver: Receiver,
+        isQualityFilter: Boolean,
+        reverseTheFilter: Boolean = false
+    ): Report {
+        // First, only do detailed logging on qualityFilters.
+        // But, **don't** do detailed logging if reverseTheFilter is true.
+        // This is a hack, but its because the logging is nonsensical if the filter is reversed.
+        // (Its nontrivial to fix the detailed logging of a reversed filter, per deMorgan's law)
+        val doDetailedFilterLogging = isQualityFilter && !reverseTheFilter
```

Comment:
The individual filters inside the qualityFilter are blissfully unaware that they might get reversed.     So the wonderful logging I just added is utterly backward in the case where the filter is reversed.